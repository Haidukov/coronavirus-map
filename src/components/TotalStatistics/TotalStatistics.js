import React from 'react';
import classNames from 'classnames';
import styles from './TotalStatistics.module.css';

const TotalStatistics = ({ totalConfirmed, totalRecovered, totalDeaths }) => {
    return (
        <div className={styles.container}>
            <div className={classNames(styles.cell, styles.confirmed)}>
                <div className={styles.title}>Confirmed</div>
                <div data-testid='confirmed'>{totalConfirmed}</div>
            </div>
            <div className={classNames(styles.cell, styles.recovered)}>
                <div className={styles.title}>Recovered</div>
                <div data-testid='recovered'>{totalRecovered}</div>
            </div>
            <div className={classNames(styles.cell, styles.deaths)}>
                <div className={styles.title}>Deaths</div>
                <div data-testid='deaths'>{totalDeaths}</div>
            </div>
        </div>
    );
};

export default TotalStatistics;