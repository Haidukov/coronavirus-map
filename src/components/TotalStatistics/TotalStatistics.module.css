.container {
   
    display: flex;
}

.cell {
    flex-grow: 1;
    padding: 10px;
    margin: 1px;
}

.title {
    font-weight: bold;
    font-size: 16px;
}

.deaths {
    background-color: red;
    color: white
}

.recovered {
    background-color: green;
    color: white;
}

.confirmed {
    background-color: yellow;
}