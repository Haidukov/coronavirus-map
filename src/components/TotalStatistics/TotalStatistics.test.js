import React from 'react';
import { render } from '@testing-library/react';
import TotalStatistics from './TotalStatistics';

describe('<TotalStatistics/>', () => {
    it('should render component', () => {
        const { container } = render(<TotalStatistics/>);
        expect(container).toMatchSnapshot();
    });

    it('should render data', () => {
        const { container, getByTestId } = render(
            <TotalStatistics 
                totalConfirmed={10000} 
                totalDeaths={1000}
                totalRecovered={2000}
            />);

        const confirmed = getByTestId('confirmed');
        const deaths = getByTestId('deaths');
        const recovered = getByTestId('recovered');

        expect(confirmed).toHaveTextContent(10000);
        expect(deaths).toHaveTextContent(1000);
        expect(recovered).toHaveTextContent(2000);
        expect(container).toMatchSnapshot();
    })
});