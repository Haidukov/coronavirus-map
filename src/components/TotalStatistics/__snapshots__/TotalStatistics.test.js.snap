// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalStatistics/> should render component 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="cell confirmed"
    >
      <div
        class="title"
      >
        Confirmed
      </div>
      <div
        data-testid="confirmed"
      />
    </div>
    <div
      class="cell recovered"
    >
      <div
        class="title"
      >
        Recovered
      </div>
      <div
        data-testid="recovered"
      />
    </div>
    <div
      class="cell deaths"
    >
      <div
        class="title"
      >
        Deaths
      </div>
      <div
        data-testid="deaths"
      />
    </div>
  </div>
</div>
`;

exports[`<TotalStatistics/> should render data 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="cell confirmed"
    >
      <div
        class="title"
      >
        Confirmed
      </div>
      <div
        data-testid="confirmed"
      >
        10000
      </div>
    </div>
    <div
      class="cell recovered"
    >
      <div
        class="title"
      >
        Recovered
      </div>
      <div
        data-testid="recovered"
      >
        2000
      </div>
    </div>
    <div
      class="cell deaths"
    >
      <div
        class="title"
      >
        Deaths
      </div>
      <div
        data-testid="deaths"
      >
        1000
      </div>
    </div>
  </div>
</div>
`;
